src/cascade/dismod/serialize.py

```diff
@@ -128,18 +129,46 @@ def make_log_table():
 def make_node_table(context):
     # Assume we have one location, so no parents.
     # If we had a hierarchy, that would be used to determine parents.
-    unique_locations = context.input_data.observations["location_id"].unique()
-    assert len(unique_locations) == 1
+    if context.input_data.observations is not None:
+        unique_locations = context.input_data.observations["location_id"].unique()
+        assert len(unique_locations) == 1
+    else:
+        warnings.warn("No observations in model, falling back to location_id in parameters")
+        unique_locations = np.array([context.parameters.location_id])
 
     return pd.DataFrame({"node_name": unique_locations.astype(int).astype(str), "parent": np.array([np.NaN])})
 
 
 def make_data_table(context):
-    observations = observations_to_data(context.input_data.observations)
-    constraints = observations_to_data(context.input_data.constraints, hold_out=1)
-    total_data = pd.concat([observations, constraints], ignore_index=True)
-    # Why a unique string name?
-    total_data["data_name"] = total_data.index.astype(str)
+    total_data = []
+    if context.input_data.observations is not None:
+        total_data.append(observations_to_data(context.input_data.observations))
+    if context.input_data.constraints is not None:
+        total_data.append(observations_to_data(context.input_data.constraints, hold_out=1))
+
```

Comment:
This looks like the central section of the PR? The idea is that there isn't any input data but there can be observations and constraints. OK.